src/storage/src/persist_cache.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! A process-global cache of persist clients to reduce contention on the SQLite consensus
+//! implementation
+
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+use std::fmt::Debug;
+
+use differential_dataflow::difference::Semigroup;
+use differential_dataflow::lattice::Lattice;
+use lazy_static::lazy_static;
+use timely::progress::Timestamp;
+use tokio::sync::Mutex;
+
+use mz_persist_client::{
+    error::InvalidUsage, read::ReadHandle, write::WriteHandle, PersistClient, PersistLocation,
+    ShardId,
+};
+use mz_persist_types::{Codec, Codec64};
+
+lazy_static! {
+    static ref CACHE: Mutex<HashMap<PersistLocation, PersistClient>> = Mutex::new(HashMap::new());
+}
```

Comment:
avoid macros if you can!
```suggestion
static CACHE: once_cell::sync::Lazy<Mutex<HashMap<PersistLocation, PersistClient>> = once_cell::sync::Lazy(|| {
    Mutex::new(HashMap::new());
}
```

Reply:
nice, switched to `once_cell`